ui/src/components/overview/MobileClientCardView.js

```diff
@@ -59,6 +57,34 @@ class MobileClientCardView extends Component {
     this.setState({ currentValue: event.target.value });
   }
 
+  getBuilds = app => {
+    const { mobileClientBuilds } = this.props;
+    const builds = mobileClientBuilds.filter(build => build.metadata.labels['mobile-client-id'] === app.metadata.name);
+    const lastBuilds = builds.reduce((prev, curr) => {
+      const bcName = curr.metadata.annotations['openshift.io/build-config.name'];
+      const buildNumber = curr.metadata.annotations['openshift.io/build.number'];
+      const buildStatus = curr.status.phase;
+      if (!prev[bcName] || prev[bcName].buildNumber < buildNumber) {
+        prev[bcName] = {
+          buildNumber,
+          buildStatus
+        };
+      }
+      return prev;
+    }, {});
+    let numFailedBuilds = 0;
+    for (const build in lastBuilds) {
+      if (lastBuilds[build].buildStatus === 'Failed') {
+        numFailedBuilds++;
+      }
+    }
+    const numInProgressBuilds = builds.filter(build => build.status.phase === 'Running').length;
```

Comment:
why the in progress number of builds is calculated from all the builds, but the failed number of builds is not? should we keep the behaviour consistent?

Reply:
The reason for showing failed builds only for last builds is because that is what is useful for user IMO. If we would show all failed builds, this number would just grow with time and there would be no value in it. On the other hand, with in-progress builds, I thing there is a value in knowing how many builds are in progress, even if there are more for one build config. WDYT?